### ISSUE TITLE
Prestats and aroma

### ISSUE THREAD

**Issue by lucyjg** [PennBBL/xcpEngine#272]
Hello, 

Forgive any stupid questions - I just want to double check a couple of things! Thank you in advance for any help you can give. 

I have done resting-state preprocessing in fmriprep (including ICA-AROMA) and now I want to finish off with denoising in xcpEngine. Having looked through some issues threads, it seems that most people have prestats like this in the design file:

###################################################################
# 1 PRESTATS
###################################################################

prestats_rerun[1]=1
prestats_cleanup[1]=1
prestats_process[1]=FMP

On the xcpEngine documention it says: prestats was previously an omnibus module for the preprocessing of functional MR images. The functionality of prestats is mostly covered by FMRIPREP. So i'm a little confused as to why 1 is set as the default and why other people who have already preprocessed in fmriprep have it set as 1 - I think I must be missing something really obvious!

Additionally, as I have performed aroma on fmriprep should I be putting confound2_aroma[2]=1 to signify that I have done this? Also, am I meant to be doing anything with the AROMAnoiseICs.csv output from fmriprep?

Many thanks in advance for any help you can give :) 

Lucy

**Comment by a3sha2** [comment]
Hi @lucyjg 
`prestats_rerun` can also be set to 0. We encourage people to set it to 1 because sometime the users may want to rerun the  same subject in the same directory, and the prestats module  read data and necessary registration files from fmriprep directory. 

if you set `confound2_aroma=1`, that means you want to include the  aroma nuisance components  in the confound regressors  to be regress out. Those aroma components  that will be included  in confound regressors  are the one in `AROMAnoiseICs.csv`.

**Comment by lucyjg** [comment]
I think I am a little confused by "rerun the same subject in the same directory" - sorry!
Additionally, is it necessary to set prestats_rerun to 1 for xcpEngine to read the necessary data and registration files from the fmriprep directory? i.e. will confound2 and regress modules work correctly with prestats_rerun set at 0?

Thank you so much for your help with this.

**Comment by a3sha2** [comment]
rerun means running the module for the second time or more. There is no problem  setting it to zero. 
setting it to 0 will not rerun the a particular module   if all the necessary outputs are already  produced  but setting it to 1 means to run it again (and overwrite the previous outputs)  even if all the required outputs are there from previous run.  

the prestats is  set to 1 in the case of rerun (for second or third time ) to make sure the inputs are the same  from  previous run.

**Comment by lucyjg** [comment]
Ok, great! Thank you very much for explaining this further. 

One final question I hope you might be able to help with: if I have used ICA-AROMA on fmriprep, when I am using xcpEngine, should I be using the sub-029_task-rest_space-MNI152NLin6Asym_desc-smoothAROMAnonaggr_bold.nii image or sub-029_task-rest_space-MNI152NLin6Asym_desc-preproc_bold.nii image - intuitively I would use the smoothed AROMA image, but I just wanted to double check :) 

Many thanks for any help you can give.

**Comment by a3sha2** [comment]
if you use `sub-029_task-rest_space-MNI152NLin6Asym_desc-smoothAROMAnonaggr_bold.nii image` you need to `confound2_aroma=0`, aroma nuisance components have been regressed out 
but  if you use `sub-029_task-rest_space-MNI152NLin6Asym_desc-smoothAROMAnonaggr_bold.nii`  image you need to set `confound2_aroma=1` if you want to regress out aroma nuisance components

**Comment by RobertJirsaraie** [comment]
@a3sha2, just to clarify on your statement above.

If we use the ICA-AROMA niftis outputted by fmriprep (`*_desc-smoothAROMAnonaggr_bold.nii.gz`) then these files already have nuisance components regressed out. So we should not enable further them to be further regressed out in the xcpengine design file (`confound2_aroma=0`). The only time when that option should be enabled is when using the regular preproc niftis outputted by fmriprep that have not yet been processed by ICA-AROMA (`*_desc-preproc_bold.nii.gz`). Correct?

I was previously denoising the ICA-AROMA niftis  `*_desc-smoothAROMAnonaggr_bold.nii.gz` with the aroma option also enabled in the xcpengine design file. This resulted in niftis with larger holes because noise components were regressed out twice if I am understanding your statement correctly. See screenshot below of denoised nifti with a red template underneath.

![Screen Shot 2020-06-25 at 11 45 59 PM](https://user-images.githubusercontent.com/29236913/85829115-bfaf1800-b73e-11ea-8dbf-8f02e6a74430.png)

**Comment by a3sha2** [comment]
@RobertJirsaraie  
Yes Robert, i think we are saying the same .  with *_desc-smoothAROMAnonaggr_bold.nii.gz , the aroma components have been regressed out and confound2_aroma=0 will remain zeros
